File: programming/4.2/main.cpp
#include <iostream>
#include <cmath>
#include <ctime>


using namespace std;

int randint(int min, int max) {
  return rand() % (max + 1 - min) + min;
}

float generateDataItem() {
  return 1.12 * randint(0, 100) - 16 * randint(0, 50);
}

void fillData(int count, float *data) {
  for (int i = 0; i < count; i++) {
    data[i] = generateDataItem();
  }
}

int findIndexMaxElement(int count, float *dataPointer) {
  int maxIndex;
  int maxValue = dataPointer[0];

  for (int i = 0; i < count; i++) {
    if (dataPointer[i] > maxValue) {
      maxValue = dataPointer[i];
      maxIndex = i;
    }
  }

  return maxIndex;
}

void showArray(int count, float *dataPointer) {
  for (int i = 0; i < count; i++) {
    cout << dataPointer[i] << endl;
  }
}

int main() {
  srand(time(NULL));

  int count = randint(10, 20);
  float *dataPointer = new float[count];

  fillData(count, dataPointer);

  cout << "Show array before swap: \n";
  showArray(count, dataPointer);

  int maxIndex = findIndexMaxElement(count, dataPointer);
  int swap;

  swap = dataPointer[4];
  dataPointer[4] = dataPointer[maxIndex];
  dataPointer[maxIndex] = swap;

  cout << "\nShow array after swap: \n";
  for (int i = 0; i < count; i++) {
    if (i == maxIndex) {
      cout << dataPointer[i] << " <<< swapred\n";
    }
    else if (i == 4) {
      cout << dataPointer[i] << " <<< swapred\n";
    }
    else {
      cout << dataPointer[i] << endl;
    }
  }

  return 0;
}

File: programming/4.2/main.py
from random import randint

def generate_data_item(i: int) -> float:
  return 1.12 * randint(-i * randint(0, 50), 100)

def fill_data(data):
  for i in range(len(data)):
    data[i] = generate_data_item(i % 10)

  return data

def find_max_index(data):
  max_value = data[0]
  max_index = None
  for i in range(len(data)):
    if data[i] > max_value:
      max_value = data[i]
      max_index = i

  return max_index

data = [None] * 20
data = fill_data(data)
swap = 0

# swap indexes
max_index = find_max_index(data)
swap = data[4]
data[4] = data[max_index]
data[max_index] = swap

print('Data before swap:')

for i in data:
  print('%.2f' % i)

print('\nData after swap:')
for i in range(len(data)):
  if i == max_index:
    print('%.2f' % data[i], ' <<< swapred')

  elif i == 4:
    print('%.2f' % data[i], ' <<< swapre')

  else:
    print('%.2f' % data[i])



